apps/sync/js/adapters/bookmarks.js

```diff
@@ -112,6 +112,11 @@ var BookmarksHelper = (() => {
     localRecord.name = remoteRecord.name;
     if (!localRecord.fxsyncRecords) {
       localRecord.fxsyncRecords = {};
+      // We remember if a record had already been created locally before we got
+      // remote data for that URL, so that we know not to remove it even when
+      // the remote data is deleted. This applies only to readonly sync, and
+      // will be removed when sync becomes read-write.
+      localRecord.createdLocally = true;
```

Comment:
We know if the collection is in readonly mode. Can we do `localRecord.createdLocally = readonly && true;` ?